podcastfy/aiengines/llm/base.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+
+from podcastfy.core.character import Character
+
+
+class LLMBackend(ABC):
+    """Abstract base class for Language Model backends."""
+    # TODO a nice mixin/helper could be made to load prompt templates from conf file (both podcast settings and character settings)
```

Comment:
exactly. llm/base.py should define common llm attributes too such as prompt_template, prompt params, prompt itself and those we have today in config.yaml

content_generator:
  gemini_model: "gemini-1.5-pro-latest" -> should replace with model_name
  max_output_tokens: 8192
  prompt_template: "souzatharsis/podcastfy_multimodal"

Reply:
So following the previous discussion, I would vouch for both an ABC and a Mixin that implements from_config ? I would love the fact that it can get and extend the the current podcastfy prompt template. Storing it in langchain hub seems to make it hard, is there a middle ground to be found?

Reply:
yes, to both but
for now, let's do the first: Mixin 
we can think about managing prompts later.